numpy/distutils/intelccompiler.py

```diff
@@ -92,9 +93,9 @@ def __init__(self, verbose=0, dry_run=0, force=0):
 
         def initialize(self, plat_name=None):
             MSVCCompiler.initialize(self, plat_name)
-            self.cc = self.find_exe('icl.exe')
-            self.lib = self.find_exe('xilib')
-            self.linker = self.find_exe('xilink')
+            self.cc = _find_exe('icl.exe')
+            self.lib = _find_exe('xilib')
+            self.linker = _find_exe('xilink')
```

Comment:
At least for new code (python 3) `_find_exe` should be spelled [`shutil.which`](https://docs.python.org/3/library/shutil.html) to avoid using "private" interfaces. If this is to be backported, then [`distutils.spawn.find_executable function`] is still a better, although seemingly undocumented, alternative.

Reply:
Since this is a backport-candidate I think I should use `distutils.spawn.find_executable`. There is also another `find_executable ` in `numpy.distutils.exec_command`. I compare results of the different `find_executable` on Monday.

Reply:
With the new commit I use `distutils.spawn.find_executable`. Since `distutils._msvccompiler` returns the original program name if no path is found, I created a wrapper around `distutils.spawn.find_executable` to ensure that the both functions are having the same results. A few tests are failing at the moment, but I can't see how my changes are causing these tests to fail. Do I miss something?